### ISSUE TITLE
Disabling pulseaudio headphone mic boost

### ISSUE THREAD

**Issue by chris-martin** [NixOS/nixpkgs#24184]
The 2016 Dell XPS 13 has a well-known issue in Linux that causes a crackling sound over the headphone jack.

As a workaround I've been running

```bash
amixer -c 0 cset 'numid=10' 1
```

although that needs to be re-run on every reboot.

The [pulseaudio mailing list](https://lists.freedesktop.org/archives/pulseaudio-discuss/2016-January/025271.html) has another suggestion: 

> you can remove the [Element Headphone Mic Boost] sections from these two files:
>
> /usr/share/pulseaudio/alsa-mixer/paths/analog-input-internal-mic.conf
> /usr/share/pulseuadio/alsa-mixer/paths/analog-input-headphone-mic.conf
>
> After that PulseAudio won't touch the boost volume any more.

Can either of the workarounds be accomplished in a NixOS config at present?

**Comment by NeQuissimus** [comment]
I put things like that in my `xserver.displayManager.sessionCommands`. Probably has no business being there but works :)

**Comment by nioncode** [comment]
@chris-martin Did you come up with a better solution than to put it into sessionCommands? It would be better to modify the PulseAudio config files, because without the noise will re-appear when switching output configurations in PulseAudio until the command is executed again.

**Comment by chris-martin** [comment]
@nioncode Nope, this is still my config: https://github.com/chris-martin/home/blob/dea92e6a8dd7e26e2ffb0d1f5e278e5af38e5595/nix/os/renzo.nix#L106-L109

**Comment by nioncode** [comment]
I'm actually using an acpid event now to fix the noise whenever headphones are plugged in:
```
  # Fix headphone white-noise on Dell XPS 13.
  services.acpid.enable = true;
  services.acpid.handlers.fixHeadphoneNoise = {
    event = "jack/headphone HEADPHONE plug";
    action = "${pkgs.alsaUtils}/bin/amixer -c0 sset 'Headphone Mic Boost' 10dB";
  };
```

**Comment by nioncode** [comment]
@chris-martin Do you still have this problem on nixos-20.03? I'm having a low volume white noise on my headphones and the `amixer` command does not change anything.

**Comment by tchebb** [comment]
The kernel bug that necessitated this workaround has been fixed since Linux 5.7, so no PulseAudio config changes should be needed anymore. See my explanation of the issue [here](https://unix.stackexchange.com/a/581386/406766).

**Comment by chris-martin** [comment]
Haven't verified, but will go ahead and close this trusting the comment that it's been fixed upstream in the kernel.
